### PR TITLE
 Fix app name validation logic and improve error message

### LINKED ISSUE #2127: Disallow creating projects with an app that starts with a number
### Describe the bug

It is possible to create projects with an app that starts with a number but this is not valid.

```
❯ briefcase new -Q app_name=1hello --no-input && cd 1hello && briefcase dev

Using override value '1hello'

[1hello] Generating a new application 'Hello World'
Using app template: https://github.com/beeware/briefcase-template, branch v0.3.21
Template branch v0.3.21 not found; falling back to development template
Using existing template (sha af397117020084a0c284c65132c0e6c1019c1b65, updated Tue Dec 31 08:20:07 2024)

[1hello] Generated new application 'Hello World'

To run your application, type:

    $ cd 1hello
    $ briefcase dev


[1hello] Installing requirements...
Requirement already satisfied: toga-gtk~=0.4.7 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (0.4.9.dev300+g1b7609dd8)
Requirement already satisfied: pytest in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (8.3.4)
Requirement already satisfied: pycairo>=1.17.0 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from toga-gtk~=0.4.7) (1.27.0)
Requirement already satisfied: pygobject>=3.50.0 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from toga-gtk~=0.4.7) (3.50.0)
Requirement already satisfied: toga-core==0.4.9.dev300+g1b7609dd8 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from toga-gtk~=0.4.7) (0.4.9.dev300+g1b7609dd8)
Requirement already satisfied: travertino>=0.3.0 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from toga-core==0.4.9.dev300+g1b7609dd8->toga-gtk~=0.4.7) (0.3.0)
Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from pytest) (1.2.2)
Requirement already satisfied: iniconfig in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from pytest) (2.0.0)
Requirement already satisfied: packaging in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from pytest) (24.2)
Requirement already satisfied: pluggy<2,>=1.5 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from pytest) (1.5.0)
Requirement already satisfied: tomli>=1 in /home/russell/.pyenv/versions/3.10.15/envs/briefcase-3.10/lib/python3.10/site-packages (from pytest) (2.2.1)
Installing dev requirements... done

[1hello] Starting in dev mode...
===========================================================================
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/russell/.pyenv/versions/3.10.15/lib/python3.10/runpy.py", line 220, in run_module
    mod_name, mod_spec, code = _get_module_details(mod_name)
  File "/home/russell/.pyenv/versions/3.10.15/lib/python3.10/runpy.py", line 146, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/home/russell/.pyenv/versions/3.10.15/lib/python3.10/runpy.py", line 157, in _get_module_details
    code = loader.get_code(mod_name)
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "<frozen importlib._bootstrap_external>", line 947, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/russell/tmp/beeware/1hello/src/1hello/__main__.py", line 1
    from 1hello.app import main
         ^
SyntaxError: invalid decimal literal

Problem running app 1hello.
```

### Steps to reproduce

Run `briefcase new -Q app_name=1hello --no-input && cd 1hello && briefcase dev`

### Expected behavior

App names that begin with a number should not be allowed.

### Screenshots

_No response_

### Environment

- Operating System: pop os 22.04
- Python version: 3.10
- Software versions:
  - Briefcase: `0.3.21.dev125+gdced62d4`

### Logs

[briefcase.2025_01_21-00_06_20.dev.log](https://github.com/user-attachments/files/18484979/briefcase.2025_01_21-00_06_20.dev.log)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by egarciamendez**
This PR fixes a bug in the is_valid_app_name function and improves the error message for better user experience.

  Changes Made

  1. Fixed Logic Bug in is_valid_app_name
  - Corrected the validation logic that was incorrectly using is_reserved_keyword(app_name)
  - Added explicit validation for the string 'None' which should be rejected as an app name
  - Updated the function docstring to accurately reflect all validation checks

  2. Enhanced Error Message
  - Replaced the single-paragraph error message with a clear, structured bullet-point format
  - Added specific validation rules that users can easily follow:
    - Not be reserved keywords (with examples)
    - Be PEP508 compliant
    - Start with a letter
    - Not end with hyphen or underscore
    - Be valid Python identifiers when hyphens are replaced with underscores

  3. Comprehensive Test Coverage
  - Added extensive test cases to test_is_valid_app_name.py covering:
    - Additional Python keywords (and, or, if, def, class, etc.)
    - Case variations of reserved words (Switch, SWITCH, Pass, PASS)
    - Invalid formatting scenarios (starts with number, ends with underscore/hyphen)
    - Edge cases (empty string, single characters)
  - Added dedicated test function test_reserved_keyword_logic_fix() to specifically verify the logic fix
  - Enhanced existing test coverage with more valid app name examples

  Problem Solved

  - Apps with reserved keyword names causing runtime issues
  - Valid app names being incorrectly rejected
  - Confusing error messages for developers

  The improved error message provides clear, actionable guidance to developers when their app name is rejected.
  
Fixes #2127

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [ ] I have read the **CONTRIBUTING.md** file
- [ ] I will abide by the code of conduct

### CODE REVIEW

**Comment by egarciamendez** [comment]
Tests failed :-( i will be looking into those when I'm back home

**Comment by egarciamendez** [comment]
Hi @freakboy3742 ! I'm getting some errors that are unrelated to my changes. At least I suppose so. Could you check those out please?

**Comment by freakboy3742** [comment]
@egarciamendez Looks like you were bitten by some transient network problems - not sure if the problem was on Github's side, or Adoptium's - but either way, re-running the tests looks like it has succeeded.

This indicates there are some tests that we should probably clean up - there's very limited situations where we should be dependent on external network availability to run the test suite, and from the look of these test failures, none of these would qualify. I've added #2409 to track this issue with the test suite.

File: docs/about/releases.rst
Comment (freakboy3742) [inline docs/about/releases.rst:1]: It looks like you've run `tox -e towncrier` here - this file shouldn't be part of the PR. We add the release notes as part of the release process.

File: src/briefcase/config.py [hunk 1]
```diff
@@ -37,7 +37,26 @@ def is_reserved_keyword(app_name):
...
 def is_valid_app_name(app_name):
-    return not is_reserved_keyword(app_name) and is_valid_pep508_name(app_name)
+    """Determine if the app name is valid.
+
+    :param app_name: The app name to validate. Checks the following:
+        - It is not a reserved keyword.
+        - It is a valid PEP508 name.
+        - Does not start with a number.
+        - Does not have a period ('.') in the name.
+        - Does not start or end with a hyphen ('-') or underscore ('_').
```
Comment (freakboy3742) [inline src/briefcase/config.py]: The explanation about validity is really a property of the *method* not the argument; so it should be part of the method docstring, not the parameter description.

```suggestion
    """Determine if the app name is valid.

    Checks the following:
        - It is not a reserved keyword.
        - It is a valid PEP508 name.
        - Does not start with a number.
        - Does not have a period ('.') in the name.

    :param app_name: The app name to validate.
```

File: src/briefcase/config.py [hunk 2]
```diff
@@ -37,7 +37,26 @@ def is_reserved_keyword(app_name):
...
+    :returns: True if the app name is valid; False otherwise.
+    """
+    return all(
+        [
+            is_valid_pep508_name(app_name),
+            not is_reserved_keyword(app_name),
+            not app_name[0].isdigit(),
+            not app_name.startswith(("-", "_")),
+            not app_name.endswith(("-", "_")),
+            app_name.replace("-", "_").isidentifier(),
```
Comment (freakboy3742) [inline src/briefcase/config.py]: Two notes here:

1. the starts/endwith `-`,`_` rules seem to be redundant, given the PEP 508 regex
2. I'm fairly sure the `isdigit()` check is also redundant, given the latter check for `isidentifier` (as an identifier can't start with a digit).

File: src/briefcase/constants.py [hunk 3]
```diff
@@ -263,6 +263,7 @@
     "nis",
     "nntplib",
     "ntpath",
+    "none",
```
Comment (freakboy3742) [inline src/briefcase/constants.py]: I'm not clear why this one is required. This list is a list of standard library modules; `none` should be caught by the `keyword.iskeyword()` check in `is_reserved_keyword()`

File: tests/commands/convert/test_input_app_name.py [hunk 4]
```diff
@@ -39,7 +39,7 @@ def test_valid_pep621_app_name(convert_command):
 
 def test_pep621_name_is_canonicalized(convert_command):
     (convert_command.base_path / "pyproject.toml").write_text(
-        '[project]\nname="test.name"', encoding="utf-8"
+        '[project]\nname="test_name"', encoding="utf-8"
```
Comment (freakboy3742) [inline tests/commands/convert/test_input_app_name.py]: This would be a regression. The PEP 621 specification for `project.name` is a PEP 508 name (because it's the name of the package being uploaded to PyPI), in which case `test.name` actually *is* legal. 

So, in this case, if the *project* has a name of `test.name`, we need to know it will be coerced to `test-name`.

File: tests/config/test_is_valid_app_name.py [hunk 5]
```diff
@@ -9,9 +9,17 @@
         "helloworld",
         "helloWorld",
         "hello42world",
-        "42helloworld",
         "hello_world",
         "hello-world",
+        "myapp",
+        "my_app",
+        "my-app",
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: I'm not sure these three are required - they're already checked by `myapp`, `hello_world` and `hello-world` above.

File: tests/config/test_is_valid_app_name.py [hunk 6]
```diff
@@ -9,9 +9,17 @@
...
+        "app2",
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: To keep the examples consistent, this should probably be `helloworld42`.

File: tests/config/test_is_valid_app_name.py [hunk 7]
```diff
@@ -9,9 +9,17 @@
...
+        "validname",
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: What is this one checking for?

File: tests/config/test_is_valid_app_name.py [hunk 8]
```diff
@@ -9,9 +9,17 @@
...
+        "a",  # single letter
+        "abc123",  # alphanumeric
+        "app-with-hyphens",
+        "app_with_underscores",
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: What are these three checking that aren't checked by previous examples?

File: tests/config/test_is_valid_app_name.py [hunk 9]
```diff
@@ -34,12 +42,66 @@ def test_is_valid_app_name(name):
...
+        "-",  # single hyphen
+        "1",  # single digit
+        "123",  # all digits
         # ı, İ and K (i.e. 0x212a) are valid ASCII when made lowercase and as such are
         # accepted by the official PEP 508 regex... but they are rejected here to ensure
         # compliance with the regex that is used in practice.
         "helloworld_ı",
         "İstanbul",
         "Kelvin",
+        # Additional Python keywords to verify logic fix
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: What "logic fix" are you referring to? Even in the context of this PR, it's not completely clear; and the context of this PR won't exist once the PR is merged. If there's a specific case that is being checked (e.g., Python identifiers), then the comment should reflect that. We also don't need to be exhaustive on every reserved word - we need to pick one or two from every *class* of possible failure (e.g., javascript reserved word, Python reserved word, python module name, etc), but we don't need to reproduce every possible bad app name.

Comment (egarciamendez) [reply tests/config/test_is_valid_app_name.py]: 'logic fix' comes from a bad translation in my head 👎 
Will fix that and remove redundant tests.

File: src/briefcase/config.py (hunk 2 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: very sharp! In hindsight those are actually redundant. I will remove those

File: src/briefcase/constants.py (hunk 3 above)
Comment (egarciamendez) [reply src/briefcase/constants.py]: It was neccesary because it wasn't being caught by .iskeyword()

Image

File: tests/commands/convert/test_input_app_name.py (hunk 4 above)
Comment (egarciamendez) [reply tests/commands/convert/test_input_app_name.py]: I'm hitting a wall right now. Checking for test.name will return False when calling is_valid_app_name and won't go on to canonicalize test.name into test-name.
afbeelding

Does that mean that a period inside the app name should be possible? 
afbeelding

Comment (freakboy3742) [reply tests/commands/convert/test_input_app_name.py]: No - `appname` must be a valid Python module. The value of `name` that comes from pyproject metadata does not have to be a valid module name.

In the case of the convert command - if you have a `project.name` of `my.app`, then that *isn't* a valid app name. `my_app` *might* be a good candidate name, though; so if `project.name` isn't a valid app name, providing "`project.name` with `.` replaced with `_`" as a candidate default name would make sense to me.

Comment (egarciamendez) [reply tests/commands/convert/test_input_app_name.py]: In that case removing the check of `name` to be a valid module will do the trick + replacing `-` with `_`

`test.name` will get converted to `test_name`

afbeelding

File: src/briefcase/config.py [hunk 10]
```diff
@@ -37,7 +37,24 @@ def is_reserved_keyword(app_name):
...
 
 def is_valid_app_name(app_name):
-    return not is_reserved_keyword(app_name) and is_valid_pep508_name(app_name)
+    """Determine if the app name is valid.
+
+    Checks the following:
+        - It is not a reserved keyword.
+        - It is a valid PEP508 name.
+        - Does not start with a number.
+        - Does not have a period ('.') in the name.
```
Comment (freakboy3742) [inline src/briefcase/config.py]: This description no longer matches the implementation.

File: src/briefcase/commands/convert.py [hunk 11]
```diff
@@ -109,15 +109,11 @@ def input_app_name(self, override_value: str | None) -> str:
...
+            "spaces or punctuation, and it can't start with a hyphen, underscore or a number."
         )
 
         default = "hello-world"
-        if (
-            "name" in self.pep621_data
-            and is_valid_app_name(self.pep621_data["name"])
-            and override_value is None
-        ):
+        if "name" in self.pep621_data and override_value is None:
```
Comment (freakboy3742) [inline src/briefcase/commands/convert.py:115]: I'm not sure why you've dropped the validity check here - `1hello.world` is a valid `project.name` by PEP 508, which canonicalizes to `1hello-world` - but that's not a valid app name, so it can't be used as a candidate.

File: src/briefcase/config.py [hunk 12]
```diff
@@ -400,9 +417,12 @@ def __init__(
         if not is_valid_app_name(self.app_name):
             raise BriefcaseConfigError(
                 f"{self.app_name!r} is not a valid app name.\n\n"
-                "App names must not be reserved keywords such as 'and', 'for' and 'while'.\n"
-                "They must also be PEP508 compliant (i.e., they can only include letters,\n"
-                "numbers, '-' and '_'; must start with a letter; and cannot end with '-' or '_')."
+                "App names must:\n"
+                "- Not be reserved keywords (like 'and', 'for', 'while', 'main', 'test', etc.)\n"
+                "- Be PEP508 compliant (letters, numbers, hyphens, and underscores only)\n"
```
Comment (freakboy3742) [inline src/briefcase/config.py]: The "PEP508 compliant" restriction isn't correct any more. `hello.world` is valid PEP 508, but it's not a valid app name.

File: src/briefcase/config.py [hunk 13]
```diff
@@ -400,9 +417,12 @@ def __init__(
...
+                "- Start with a letter (not a number, hyphen, or underscore)\n"
+                "- Not end with a hyphen or underscore\n"
```
Comment (freakboy3742) [inline src/briefcase/config.py]: This is another PEP508 discrepancy - `foo_` isn't a valid PEP 508 name, but it *is* a valid app name, because it's a valid module name (i.e., identifier).

File: src/briefcase/constants.py (hunk 3 above)
Comment (freakboy3742) [reply src/briefcase/constants.py]: Ok - in which case that points out the problem - `none` *would* be a legal app name (it would be a *daft* app name, but it would be legal 😄). However, `None` would *not* be a valid app name.

File: tests/commands/convert/test_input_app_name.py (hunk 4 above)
Comment (freakboy3742) [reply tests/commands/convert/test_input_app_name.py]: No, it won't - `1test.name` is a valid PEP 508 name that canonicalises to `1test_name`, but that isn't a valid app name.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This revision has overcorrected a little too much - it's allowing names that shouldn't be allowed, and rejecting names that should be valid. To be fair - some of those rejected names were historically rejected as well - so this is at least partially a case of the existing test cases being misleading.

I think the core of the problem is that PEP 508 compliance in the app name is a red herring. Despite the historical code, we don't actually *want* PEP 508 compliance for app names - we want "is valid identifier after replacing `-`  with `_`" and "is not a keyword".

The convert wizard then needs different handling, because `project.name` is governed *solely* by PEP 508 - but there are valid PEP508 names that aren't valid app names (e.g., `foo.bar`), and names that *aren't* valid PEP 508 names, but *are* validated app names (e.g., `foo_`). In the convert wizard, there's essentially 2 cases:

1. The project name is a valid app name as written
2. The project name can be canonicalised into a valid app name. 

Case 1 is the simple case (e.g., `project.name = "foobar"`) - in which case, the name is being used unmodified, so we can use it as the app name without any confirmation.

Case 2 is the slightly more complex case, where we are applying a transformation (e.g., `project.name = "foo.bar"`, which canonicalises to `foo_bar`). In this case, that's likely a good candidate, but we need to get the user to confirm this.

If neither of these cases are true - `1foo.bar`, for example - then we need to fall back to alternate suggestions (such as using the directory name).

Does that make sense?

File: src/briefcase/commands/convert.py (hunk 11 above)
Comment (egarciamendez) [reply src/briefcase/commands/convert.py:115]: I dropped it because otherwise I wouldn't be able to pass this test:

Image

`test.name` from the [project] name in pyproject.toml isn’t a valid app name and therefore won’t be canonicalized to `test-name`.

To be honest, I’m a bit confused right now about what I should do instead.

File: src/briefcase/config.py (hunk 12 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: done

File: src/briefcase/config.py (hunk 13 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: so ending with `-` or `_` is ok??

Comment (freakboy3742) [reply src/briefcase/config.py]: Yes - you can name a variable or module `foo_`, so it's a valid app name.

The core guidance here is it must be a valid Python identifier after any `-` has been replaced with `_`, and not be a reserved word (not just a Python reserved word, but a reserved word in any of the languages we cover).

File: src/briefcase/config.py [hunk 14]
```diff
@@ -33,11 +33,30 @@ def is_valid_pep508_name(app_name):
...
+    Checks the following:
+        - It is not a reserved keyword.
+        - It is a valid PEP508 name.
+
+    :param app_name: The app name to validate.
+    :returns: True if the app name is valid; False otherwise.
+    """
+    return all(
+        [
+            is_valid_pep508_name(app_name),
```
Comment (freakboy3742) [inline src/briefcase/config.py]: As noted in my previous review - PEP508 seems like it's a historical red herring here. PEP508 imposes restrictions that don't apply to app names (no trailing `-` or `_`), and allows things that aren't valid (starting with digits and allowing `.`). 

AFAICT, what we need her is entirely determined by "is valid identifier after converting `-` to `_`", excluding reserved words.

File: src/briefcase/commands/convert.py (hunk 11 above)
Comment (freakboy3742) [reply src/briefcase/commands/convert.py:115]: There's 3 cases here:

1. The project name is useable as-is. `foobar` and `foo-bar` are both valid project names, and are also valid app names as-is. They can be directly provided as the candidate app name, as there's no transformation being applied. 
2. The project name *could* be a valid name, but requires some canonicalisation. `test.name` is a valid PEP 508 project name, and it canonicalizes to `test-name`; we don't want to provide that name as an *automatic* app name, but it's a reasonable candidate for an app name that the user could confirm as valid.
3. The project name *isn't* valid, even after canonicalisation. `42foobar` *is* a valid project name - but it canonicalizes to exactly the same value, which isn't a valid app name. In this case, we need to fall back to other hinted options (such as the base path of the project).

The code has a branch for 1 and 3; but the logic that triggers 1 is incorrect and it merges in the canonicalisation required by case 2. There's no logic handling the case of "test.name" being coerced into a *hint* for the user to confirm.

Does that make more sense?

**Review by freakboy3742 — CHANGES_REQUESTED** [review]

File: src/briefcase/config.py (hunk 13 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: The fix should be to remove the is_valid_pep508_name(app_name) check from is_valid_app_name(), since the real requirement is it to be a valid Python identifier after dash-to-underscore conversion and not a reserved word. I will fix that

File: src/briefcase/commands/convert.py (hunk 11 above)
Comment (egarciamendez) [reply src/briefcase/commands/convert.py:115]: Yes, thank you for the explanation (and your patience 😅)

File: tests/config/test_is_valid_app_name.py [hunk 15]
```diff
@@ -9,9 +9,14 @@
         "helloworld",
         "helloWorld",
         "hello42world",
-        "42helloworld",
         "hello_world",
         "hello-world",
+        "helloworld42",
+        "a",  # single letter
+        "abc123",  # alphanumeric
+        "none",  # daft but legal ;-)
```
Comment (freakboy3742) [inline tests/config/test_is_valid_app_name.py]: Worth clarifying exactly why a word that looks like a reserved word is actually legal
```suggestion
        "none",  # `None` is illegal, but when lower case, it isn't a reserved word
```

File: src/briefcase/commands/convert.py (hunk 11 above)
Comment (freakboy3742) [reply src/briefcase/commands/convert.py:115]: Ok - but my the original point still stands: you've dropped the validation check on this branch. By this code, if user provides *any* value for `name` in the `[project]` data, the canonicalised version of that name *will* be used. You've added code for the three cases that I described - but this check comes before those cases are evaluated.

File: src/briefcase/config.py [hunk 16]
```diff
@@ -26,18 +26,43 @@
 PEP508_NAME_RE = re.compile(r"^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9])$")
```
Comment (freakboy3742) [inline src/briefcase/config.py]: This regex isn't being used any more, so it can be dropped.

File: src/briefcase/config.py [hunk 17]
```diff
@@ -26,18 +26,43 @@
...
+    :returns: True if the app name is valid; False otherwise.
+    """
+    module_name = app_name.lower().replace("-", "_")
+
+    # ı, İ and K (i.e. 0x212a) are valid ASCII when made lowercase and as such are
+    # accepted by the official PEP 508 regex... but they are rejected here to ensure
+    # compliance with the regex that is used in practice.
+    problematic_unicode = {"\u0131", "\u0130", "\u212a"}  # ı, İ, K
+    if any(char in app_name for char in problematic_unicode):
+        return False
```
Comment (freakboy3742) [inline src/briefcase/config.py]: The "problematic chars" handling is a hangover of PEP508 - if we're not adhering to PEP508, we don't need to preserve this behavior either. The three characters called out here are valid in Python identifier names.

File: src/briefcase/commands/convert.py [hunk 18]
```diff
@@ -109,15 +109,11 @@ def input_app_name(self, override_value: str | None) -> str:
             "We need a name that can serve as a machine-readable Python package name for "
             "your application. This name must be PEP508-compliant - that means the name "
```
Comment (freakboy3742) [inline src/briefcase/commands/convert.py]: The name doesn't need to be PEP 508 compliant, so this text can be updated.

File: src/briefcase/commands/convert.py [hunk 19]
```diff
@@ -126,13 +122,28 @@ def input_app_name(self, override_value: str | None) -> str:
...
+        ):
+            default = self.base_path.name
+            intro += (
+                "\n\n"
+                f"Based on your PEP508 formatted directory name, we suggest an "
+                f"app name of '{default}', but you can use another name if you want."
+            )
+        # Case 2: Project name could be valid after canonicalization (e.g., "test.name" -> "test-name", "test-app_name" -> "test-app-name")
+        elif is_valid_app_name(canonicalized_name):
+            default = canonicalized_name
```
Comment (freakboy3742) [inline src/briefcase/commands/convert.py:147]: By this logic, case 1 and case 2 are functionally identical. The key difference between case 1 and case 2 is that we don't need to ask the user to confirm case 1 - we're using *exactly* the same name, so it's fine as is. Case 2 involves a transformation, so it needs to be confirmed.

File: src/briefcase/config.py (hunk 16 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: Actually it is being used in some android sdk dependency. That's why I didn't delete it.

File: src/briefcase/config.py (hunk 17 above)
Comment (egarciamendez) [reply src/briefcase/config.py]: Will remove those asap

File: src/briefcase/commands/convert.py (hunk 18 above)
Comment (egarciamendez) [reply src/briefcase/commands/convert.py]: Will fix it soon

File: src/briefcase/config.py (hunk 16 above)
Comment (freakboy3742) [reply src/briefcase/config.py]: Ah - I somehow missed that usage in my searching. 

In this case, I think that's a case where the regex can be moved and renamed. It's being used to identify valid Android emulator names, which aren't bound by PEP 508 - they just happen to have very similar rules. 

Interestingly, PEP 508 will allow `bee.phone` as a name, and that *works* with the emulator (I just checked), but period isn't included in the description of a valid emulator name. 

So - I'd suggest moving it to the android module where it's being used, and rename it to `ANDROID_EMULATOR_NAME_RE`.

Comment (egarciamendez) [reply src/briefcase/config.py]: Done :-)

File: src/briefcase/commands/convert.py (hunk 19 above)
Comment (egarciamendez) [reply src/briefcase/commands/convert.py:147]: 👍

**Comment by freakboy3742** [comment]
Are you waiting on a review of this code? Aside from the CI pre-commit failure, it looks like some of my comments from my previous review are outstanding. When you're ready for another review, let me know and I'll take a look.
